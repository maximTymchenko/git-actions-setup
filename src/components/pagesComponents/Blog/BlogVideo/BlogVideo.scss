@import '../../../../styles/vars';

.blog-video {
    outline: 2px solid $secondary;
    box-shadow: 9px 9px 1px 1px rgba(57, 156, 203, 0.4);
    @include transition__outline;
    @include transition__shadow;
    position: relative;
    cursor: pointer;

    &:hover {
        outline: 4px solid $secondary;
        outline-offset: -2px;
    }

    &:active {
        @include button__press;
        box-shadow: 9px 9px 1px 1px rgba(57, 156, 203, 0);
    }

    &::after {
        content: '';
        position: absolute;
        background-color: $primary__light--80;
        opacity: 0.9;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
    }

    &__background {
        position: absolute;
        top: 0;
        right: 0;
        left: 0;
        bottom: 0;
        width: 100%;
        height: 100%;
        overflow: hidden;
    }

    &__video {
        position: relative;
        bottom: 50px;
        right: 80px;

        &--aws {
            bottom: 150px;
            // top: 140px;
            right: 330px;
        }
    }

    &__content {
        width: 615px;
        height: 360px;
        position: relative;
        z-index: 20;

        display: grid;
        grid-template-columns: 1fr;
    }

    &__head {
        display: flex;
        justify-content: space-between;
    }

    &__asside {
        display: flex;
        flex-direction: column;
        margin-top: 10px;
        margin-right: 10px;
    }

    &__date {
        opacity: 0.88;
        font-family: 'OktaNeue-Black';
        font-size: 10px;
        font-weight: 900;
        line-height: 1.1;
        letter-spacing: 0.5px;
        text-align: right;
        color: $background__primary;
        margin-bottom: 6px;
    }

    &__time {
        color: $background__primary;
        background-color: #000;
        font-family: 'OktaNeue-Bold';
        font-size: 14px;
        font-weight: bold;
        line-height: 1.71;
        text-align: center;
        align-self: flex-end;
        margin-bottom: 10px;
        display: block;
        width: 40px;
    }

    &__slogan {
        font-size: 26px;
        font-weight: bold;
        line-height: 1.5;
        color: $background__primary;
        font-family: 'OktaNeue-Bold';
        margin-left: 5px;
        align-self: end;
    }

    &__footer {
        display: flex;
        margin-top: 15px;
    }

    &__hashtag {
        font-family: 'OktaNeue-Black';
        font-size: 11px;
        font-weight: 900;
        letter-spacing: 1px;
        color: $text__body;
    }

    &__label {
        background-color: #000;
        color: $background__primary;
        font-family: 'OktaNeue-Black';
        font-size: 10px;
        font-weight: 900;
        line-height: 1.1;
        letter-spacing: 0.5px;
        display: inline-block;
        padding: 2px 3px;
        height: 17px;
        margin-right: 5px;
    }
}
