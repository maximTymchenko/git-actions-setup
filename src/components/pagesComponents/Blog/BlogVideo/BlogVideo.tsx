import React, { FC } from 'react';
import './BlogVideo.scss';
import Button from '../../../common/Button';
import triangleRight from '../../../../assets/img/triangle-right.svg';
import video from '../../../../assets/video/meeting_video.mp4';

interface IVideoBlogProps {
    videoDesc: string;
    video: video;
    videoPos: string;
    size: string;
}

const BlogVideo: FC = (props: IVideoBlogProps) => {
    return (
        <div>
            <div data-testid="blog-video" className="blog-video">
                <div className="blog-video__background">
                    <video
                        className={`blog-video__video blog-video__video--${props.videoPos}`}
                        src={props.video}
                        autoPlay
                        muted
                        loop
                    />
                </div>
                <div className={`blog-video__content blog-video__content--${props.size}`}>
                    <div className="blog-video__head">
                        <Button type="watch" desc="Watch now" icon={triangleRight}>
                            Watch now
                        </Button>
                        <div className="blog-video__asside">
                            <span className="blog-video__time">15:10</span>
                            <span className="blog-video__date">Feb 27, 2013</span>
                        </div>
                    </div>
                    <p className="blog-video__slogan">{props.videoDesc}</p>
                </div>
            </div>
            <div className="blog-video__footer">
                <span className="blog-video__label">VIDEO</span>
                <p className="blog-video__hashtag">
                    Microservice architecture, Microsoft HoloLens 2, REST API, Web development, +13
                    more
                </p>
            </div>
        </div>
    );
};

export default BlogVideo;
