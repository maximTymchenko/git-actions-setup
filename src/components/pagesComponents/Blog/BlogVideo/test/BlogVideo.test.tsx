import React from 'react';
import BlogVideo from '../BlogVideo';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import video from '../../../../../assets/video/meeting_video.mp4';

it('renders elem correctly', () => {
    const { getByTestId } = render(
        <BlogVideo videoDesc="Hiring the best possible engineers" video={video} />,
    );
    expect(getByTestId('blog-video')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer
        .create(<BlogVideo videoDesc="Hiring the best possible engineers" video={video} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
