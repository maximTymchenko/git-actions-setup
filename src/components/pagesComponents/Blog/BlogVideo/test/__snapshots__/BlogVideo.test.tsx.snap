// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div>
  <div
    className="blog-video"
    data-testid="blog-video"
  >
    <div
      className="blog-video__background"
    >
      <video
        autoPlay={true}
        className="blog-video__video blog-video__video--undefined"
        loop={true}
        muted={true}
        src="test-file-stub"
      />
    </div>
    <div
      className="blog-video__content blog-video__content--undefined"
    >
      <div
        className="blog-video__head"
      >
        <button
          className="button button--watch button--undefined button--undefined"
          data-testid="button"
        >
          Watch now
          <img
            alt="Watch now"
            className="button__icon button__icon--undefined"
            data-testid="button__icon"
            src="test-file-stub"
          />
        </button>
        <div
          className="blog-video__asside"
        >
          <span
            className="blog-video__time"
          >
            15:10
          </span>
          <span
            className="blog-video__date"
          >
            Feb 27, 2013
          </span>
        </div>
      </div>
      <p
        className="blog-video__slogan"
      >
        Hiring the best possible engineers
      </p>
    </div>
  </div>
  <div
    className="blog-video__footer"
  >
    <span
      className="blog-video__label"
    >
      VIDEO
    </span>
    <p
      className="blog-video__hashtag"
    >
      Microservice architecture, Microsoft HoloLens 2, REST API, Web development, +13 more
    </p>
  </div>
</div>
`;
